kong/plugins/file-log/handler.lua

```diff
@@ -27,49 +30,41 @@ local file_descriptors = {}
 -- @param `premature` see OpenResty `ngx.timer.at()`
 -- @param `conf`     Configuration table, holds http endpoint details
 -- @param `message`  Message to be logged
-local function log(premature, conf, message)
-  if premature then
-    return
-  end
-
+local function log(conf, message)
   local msg = cjson.encode(message) .. "\n"
 
   local fd = file_descriptors[conf.path]
 
   if fd and conf.reopen then
     -- close fd, we do this here, to make sure a previously cached fd also
     -- gets closed upon dynamic changes of the configuration
-    ffi.C.close(fd)
+    C.close(fd)
     file_descriptors[conf.path] = nil
     fd = nil
   end
 
   if not fd then
-    fd = ffi.C.open(conf.path, oflags, mode)
+    fd = C.open(conf.path, oflags, mode)
     if fd < 0 then
       local errno = ffi.errno()
-      ngx.log(ngx.ERR, "[file-log] failed to open the file: ", ffi.string(ffi.C.strerror(errno)))
+      ngx.log(ngx.ERR, "[file-log] failed to open the file: ", ffi.string(C.strerror(errno)))
     else
       file_descriptors[conf.path] = fd
     end
   end
 
-  ffi.C.write(fd, msg, #msg)
+  C.write(fd, msg, #msg)
 end
 
 local FileLogHandler = {}
 
 FileLogHandler.PRIORITY = 9
-FileLogHandler.VERSION = "2.0.0"
+FileLogHandler.VERSION = "2.0.1"
 
 function FileLogHandler:log(conf)
-  local message = basic_serializer.serialize(ngx)
-
-  local ok, err = ngx_timer(0, log, conf, message)
-  if not ok then
-    ngx.log(ngx.ERR, "[file-log] failed to create timer: ", err)
-  end
+  local message = serialize(ngx)
 
+  log(conf, message)
```

Comment:
since this is (for good reasons) deviating from the other log plugin implementations, should we add a small comment as to why there are no timers here?

Reply:
I would say that seems to be unnecessary. There is no "standard" implementation for log plugins and for the most part this PR is trying to fix an unintentional copy-n-paste error, which shouldn't warrant a comment to explain why.

If anything, we should comment the ones that created the timer instead to explain why they are necessary in the first place.

Reply:
> fix an unintentional copy-n-paste error, which shouldn't warrant a comment to explain why.

Wouldn’t the presence of this error itself imply that a comment is warranted?